java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -1246,12 +1248,22 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_makeCudfColumnView(
       ret.reset(new cudf::column_view(cudf::data_type{cudf::type_id::STRING}, size, nullptr, valid,
                                       j_null_count, 0, {offsets_column, data_column}));
     } else if (n_type == cudf::type_id::LIST) {
-         JNI_NULL_CHECK(env, j_offset, "offset is null", 0);
-         cudf::column_view *child_view = reinterpret_cast<cudf::column_view *>(j_child);
-         cudf::size_type *offsets = reinterpret_cast<cudf::size_type *>(j_offset);
-         cudf::column_view offsets_column(cudf::data_type{cudf::type_id::INT32}, size + 1, offsets);
-         ret.reset(new cudf::column_view(cudf::data_type{cudf::type_id::LIST}, size, nullptr, valid,
+      JNI_NULL_CHECK(env, j_offset, "offset is null", 0);
+      cudf::jni::native_jpointerArray<cudf::column_view> children(env, j_children);
+      cudf::column_view *child_view = reinterpret_cast<cudf::column_view *>(children[0]);
+      cudf::size_type *offsets = reinterpret_cast<cudf::size_type *>(j_offset);
+      cudf::column_view offsets_column(cudf::data_type{cudf::type_id::INT32}, size + 1, offsets);
+      ret.reset(new cudf::column_view(cudf::data_type{cudf::type_id::LIST}, size, nullptr, valid,
                                                  j_null_count, 0, {offsets_column, *child_view}));
+   } else if (n_type == cudf::type_id::STRUCT) {
+     cudf::jni::native_jpointerArray<cudf::column_view> children(env, j_children);
+     std::vector<column_view> children_vector(children.size());
+     for (int i = 0; i < children.size(); i++) {
+       cudf::column_view *child_view = reinterpret_cast<cudf::column_view *>(children[i]);
```

Comment:
again the reinterpret cast is not needed you should be able to do this as a single line.

```
children_vector[i] = *children[i];
```